### PR TITLE
support for Operators which calcite dont support

### PR DESCRIPTION
**Pull request by kanha-gupta**
Ref #24200

I have added support for Operators through Calcite method calls which calcite don't support natively.
SqlCaseIDs ;-
1.select_where_with_bit_expr_with_ampersand
2.select_where_with_bit_expr_with_caret
3.select_where_with_bit_expr_with_div
4.select_where_with_bit_expr_with_signed_left_shift
5.select_where_with_bit_expr_with_signed_right_shift

### CODE REVIEW

**Comment by kanha-gupta** [comment]
@strongduanmu Please review :)

**Comment by kanha-gupta** [comment]
@strongduanmu Respected sir, please review for any changes  :)

File: kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/BinaryOperationExpressionConverter.java [hunk 1]
```diff
@@ -72,6 +72,11 @@ private static void register() {
         register(SqlStdOperatorTable.CONCAT);
         register(SqlStdOperatorTable.PATTERN_ALTER);
         register(SqlStdOperatorTable.MOD);
+        register(OperatorConverter.DIV);
```
Comment (strongduanmu) [inline kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/BinaryOperationExpressionConverter.java:75]: Can you rename this calss to SQLExtensionOperatorTable?

File: kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/OperatorConverter.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sqlfederation.optimizer.converter.segment.expression.impl;
+
+import org.apache.calcite.sql.SqlBinaryOperator;
+import org.apache.calcite.sql.SqlKind;
+
+public final class OperatorConverter {
```
Comment (strongduanmu) [inline kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/OperatorConverter.java:23]: Please add java doc for this calss.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by kanha-gupta** [comment]
@strongduanmu Changes done, please check :)

File: kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/BinaryOperationExpressionConverter.java (hunk 1 above)
Comment (kanha-gupta) [reply kernel/sql-federation/optimizer/src/main/java/org/apache/shardingsphere/sqlfederation/optimizer/converter/segment/expression/impl/BinaryOperationExpressionConverter.java]: Done

**Comment by kanha-gupta** [comment]
@strongduanmu Please merge this if errors are resolved.

**Comment by strongduanmu** [comment]
> @strongduanmu Please merge this if errors are resolved.

Hi @kanha-gupta, please solve code conflict and ci error.

**Comment by kanha-gupta** [comment]
@strongduanmu Please review, errors are resolved now

**Review by strongduanmu — APPROVED** [review]
LGTM, merged.

**Pull request merged**